laurent22/joplin
# Desktop: Fixes #9826: Fix drag-and-drop of images and text in the rich text editor

# Summary

Fixes #9826 by only calling `preventDefault` in `ondrop` when Joplin handles the event.

# Testing

1. Create a new note in the rich text editor
2. Add a few paragraphs
3. Select the last paragraph and drag it to the start of the first
4. Add an image
5. Select the image and drag it to a new location
6. Drop a `.txt` or `.mid` file into the editor (regression test for https://github.com/laurent22/joplin/issues/5268)
    - verify that a link is created and that the file can be opened with <kbd>ctrl</kbd>-<kbd>click</kbd>
7. Drop another note into the editor
    - verify that a link is added at the cursor location and can be opened with <kbd>ctrl</kbd>-<kbd>click</kbd>

This has been successfully tested on MacOS Sonoma.
